### ISSUE TITLE
Manage localization via cooperation system

### ISSUE THREAD

**Issue by Bluefissure** [quisquous/cactbot#1255]
## Summary

We can use a cooperation system like [crowdin](https://crowdin.com/) or [transifex](https://www.transifex.com/) to cooperate with the localizations of responses and timeline texts of **raidboss** module. (Or even the whole project?)

Then we can make it easier for those who cannot code well to help with the translation (without modifying the js file).

These two system are both free for open-source projects ([crowdin](https://crowdin.com/page/open-source-project-setup-request), [transifex](https://docs.transifex.com/projects/open-source-project)).

## Current Progress

I have a demo project to export/import the responses/timelines from/to the js files, which is mostly designed for CN localizations:
- Project: [raidboss_translator](https://github.com/Bluefissure/raidboss_translator)
- Sheet: [exported sheet](https://docs.qq.com/sheet/DSFRGa1V1d2pOZk1H?tab=qwmxvl&c=E42A0RF0)

Do you think it's necessary and more efficient for later maintenance?

**Comment by quisquous** [comment]
Sorry for the delay in responding here.

I have to admit that I am a little skeptical of the overhead of this approach.  Do you feel that there are many people who would be contributing to cactbot translations but aren't because they don't know JavaScript? I would hope that it is not too hard to translate things in JavaScript files, but I am somewhat biased here because I am a programmer.  But, I think the presence of somebody like VanSlanzar who is not a programmer speaks well to other people being able to pick it up and copy surrounding code.

One very big downside to having things in a file is that there is already confusion about what a particular mechanic means, and having translations in separate sheets and projects farther away from the code will make it harder to understand what "dorito stack" or "pepperoni" mean.

I think the hard part of cactbot translations is the replacement restrictions (collisions, tests, etc) and general regex knowledge.  I don't think a separate translation sheet will help with either of these problems.

One thing that I think has helped translations is to have `resources/responses.js` as well as `ui/raidboss/common_translation.js`.  I think another thing that will help along the lines of what @MaikoTan asked for is to do something like this https://github.com/quisquous/cactbot/issues/1307#issuecomment-608283837 for common translations.  I personally don't think it's worth doing this for absolutely everything, but just things that get used multiple times.  That will just cut down on the total number of translations that are needed overall.

Sorry for the skepticism, but I am just not convinced that an approach like this solves enough problems in my mind to be worth the extra cost to do it.

**Comment by Bluefissure** [comment]
Yes, this approach is more useful to translate old stuff since most of them don't have translations for cn.
Now cactbot is more frequently used by not only the EU/NA but also players & devs from other clients, so updating the translation as patch releasing is enough.

**Comment by JLGarber** [comment]
I'm in agreement with quisquous here. The solutions noted in the OP (to my mind at least) seem more geared toward projects with a constantly expanding base of content requiring translation. For cactbot, the "base translated content" is going to be relatively slow to update, and the benefit of these outside solutions would be small. In this situation, I think the solution MaikoTan and I proposed might take less overall overhead. The way such a file would be structured should be reasonably understandable even by non-programmers with a little explanation. Obviously translations that take parameters might not be so easy, but simple stuff like "Stack With Partner" ought to be doable for just about anyone.

(Additionally, if in the future we do need to convert over to a dedicated translation CMS, it should in theory be simple enough to extract the translation functions and returned objects.)

**Comment by MaikoTan** [comment]
@quisquous  I want to recall this issue because I found that [crowdin](https://crowdin.com/page/open-source-project-setup-request) they are kind, and we can easy to do a free open source request. (I have sent some emails to ask them and they said only if the repo owner would complete the request form, then they could offer this service.)

At lease we can host our all the documents onto crowdin and localize it easier?

**Comment by quisquous** [comment]
Thanks for bringing this up again.  I think I am still not sure that something like this is a good fit for raidboss translations, but doc translations are maybe a different story.  I will take another look at these, thank you!

**Comment by quisquous** [comment]
I also looked at gitlocalize a little bit as well.  I think that might be a better fit for what I am looking for (keeping markdown doc translations up to date, more lightweight).  crowdin and transifex seem like huge enterprise solutions that I need to look closer at to understand how they would fit well.

**Comment by MaikoTan** [comment]
Just found that [crowdin](https://crowdin.com/) renewed there pricing and now you can host unlimited strings on it with a free tier.
Compared different localisation website, I still recommend `crowdin` for localising the documents.
